hal/common/SerialBase.cpp

```diff
@@ -26,10 +26,11 @@ SerialBase::SerialBase(PinName tx, PinName rx) :
                                                  _thunk_irq(this), _tx_usage(DMA_USAGE_NEVER),
                                                  _rx_usage(DMA_USAGE_NEVER),
 #endif
-                                                _serial(), _baud(9600) {
+                                                _serial(), _baud(MBED_CONF_CORE_STDIO_BAUD_RATE) {
```

Comment:
Whatever you want it to be! :) Right now it's 9600:

https://github.com/ARMmbed/mbed-os/blob/master/mbed_lib.json#L11